js/gacha.js

```diff
@@ -86,9 +95,23 @@ function pullCards(slots) {
     //This applies to as many rarities as declared in the slot.
     for (let rarity in slot) {
       if (dice <= slot[rarity]) {
-        card = getRandomCards(cards_by_rarity[rarity], 1)[0];
+        pulled_cards = getRandomCards(cards_by_rarity[rarity], 4);
```

Comment:
To check my understanding: for each card pull, pick 4 random cards. Then among those 4 cards, pick the first non-dupe.
In the slim chance occurs that all 4 pulled_cards are already in `cards`, the dupe will get added to the 10-pull? Maybe add a comment about that

Reply:
Correct. The chance of the 4 cards being duplicates is extremely slim, virtually impossible, but if it ever happens it's better to return the duplicate rather than risk an infinite reroll situation.